### PR TITLE
Feature/sdl 0192 button subscription response from hmi

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
Guidelines updates according to the proposal [SDL-0192 Button Subscription response from HMI](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0192-button_subscription_response_from_hmi.md) 

This PR is **ready** for review.

### Summary
add SubscribeButton, UnsubscribeButton index, assets; upd config.json

### CODE REVIEW

File: docs/Buttons/UnsubscribeButton/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+## UnsubscribeButton
+
+Type
+: Function  
+
+Sender
+: SDL
+
+Purpose
+: Unsubscribe mob app from a button
```
Comment (ashwink11) [inline docs/Buttons/UnsubscribeButton/index.md:10]: @KhrystynaDubovyk Could you please correct the typo "mob app"?

File: docs/Buttons/SubscribeButton/index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+## SubscribeButton
+
+Type
+: Function  
+
+Sender
+: SDL
+
+Purpose
+: Subscribe mob app to a particular button
```
Comment (ashwink11) [inline docs/Buttons/SubscribeButton/index.md:10]: @KhrystynaDubovyk Could you please correct the typo "mob app"?

**Review by ashwink11 — CHANGES_REQUESTED** [review]

File: docs/Buttons/UnsubscribeButton/index.md (hunk 1 above)
Comment (KhrystynaDubovyk) [reply docs/Buttons/UnsubscribeButton/index.md]: @ashwink11, please see https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/253/commits/c61ac02ccc8d459f937fbc67c68cf7734ccad09f

File: docs/Buttons/SubscribeButton/index.md (hunk 2 above)
Comment (KhrystynaDubovyk) [reply docs/Buttons/SubscribeButton/index.md]: @ashwink11, please see https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/253/commits/c61ac02ccc8d459f937fbc67c68cf7734ccad09f

**Comment by ashwink11** [comment]
@jordynmackool this PR is ready for review and is approved by Ford.

File: config.json [hunk 3]
```diff
@@ -1,585 +1,588 @@
-{
```
Comment (iCollin) [inline config.json:1]: Please revert all changes to this file besides the addition of the two new RPCs and the removal of `OnButtonSubscription` notification

**Review by iCollin — CHANGES_REQUESTED** [review]
Hello, please remove the folder `Buttons/OnButtonSubscription`.

In both of the new RPC files, formatting changes are needed. Please compare these two files to other RPC request/response pairs to see what these new files should look like.

**Comment by KhrystynaDubovyk** [comment]
> Hello, please remove the folder `Buttons/OnButtonSubscription`.
> 
> In both of the new RPC files, formatting changes are needed. Please compare these two files to other RPC request/response pairs to see what these new files should look like.

hello @iCollin. thanks for the suggestion regarding formatting changes.
could you please advise what "other RPC request/response" that I should use as a reference?

For example, I referred https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/blob/master/docs/AppService/GetAppServiceData/index.md
here note section is written this way 
```
!!! note

* **HMI->SDL** if HMI App Service Consumer (ASC) is requesting service data
* **SDL->HMI** if HMI App Service Provider (ASP) is receiving a service data request

!!!
```
but then in the same interface note section looks like this 
https://raw.githubusercontent.com/smartdevicelink/sdl_hmi_integration_guidelines/master/docs/AppService/AppServiceActivation/index.md
```
!!! NOTE
SDL will:
1. Activate the service with ID `serviceID` if `activate` is set to true and the service is currently inactive. This will deactivate the current active service of its type.
2. Deactivate the service with ID `serviceID` if `activate` is set to false and the service is currently active. If the <abbr title="App Service Consumer">ASC</abbr> is a mobile application, the embedded service for this type will be activated in its place if available.
!!!
```
also, in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/blob/master/docs/AppService/GetAppServiceData/index.md
```
Type
: Request
```

but in https://raw.githubusercontent.com/smartdevicelink/sdl_hmi_integration_guidelines/master/docs/AppService/AppServiceActivation/index.md
```
Type
: Function
```

I am aware of the spec https://github.com/smartdevicelink/sdl_markdown_spec/tree/master, 
but still, please advise what RPC request/response I should use as a reference (because while writing this, I also referred to some of the existing index.md, which apparently was not the best one).

**Comment by iCollin** [comment]
Any other Request / Response pair should be fine to look at as an example. 

My issues are not with the small things you point out. Sorry to not be specific. My issues are the missing section header `### JSON Message Examples` and following invalid indentation, extra whitespace after some headers, missing newlines before/after headers. I see these problems in both `Buttons.SubscribeButton`/`Buttons.UnsubscribeButton` files. 

Above comments are still outstanding. Thank you.

File: docs/Buttons/SubscribeButton/index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+Type
+: Function  
+
+Sender
+: SDL
+
+Purpose
+: Subscribe application to a particular button
+
+### Request
```
Comment (iCollin) [inline docs/Buttons/SubscribeButton/index.md:12]:
```suggestion
### Request

```

File: docs/Buttons/SubscribeButton/index.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+|buttonName|[Common.ButtonName](../../common/enums/#buttonname)|true||
+
+### Response
+
+!!! MUST
+1. Check whether the button subscription can be performed for requested `buttonName`.
+2. Respond to SDL within DefaultTimeout period after successful button subscription.
+!!!
+
+#### Parameters
```
Comment (iCollin) [inline docs/Buttons/SubscribeButton/index.md:33]:
```suggestion
#### Parameters

```

File: docs/Buttons/SubscribeButton/index.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+#### Example Request
+
+```json
+{
+  "id": 32,
+  "jsonrpc": "2.0",
+  "method": "Buttons.SubscribeButton",
+  "params": {
+    "appID": "680015438",
```
Comment (iCollin) [inline docs/Buttons/SubscribeButton/index.md:52]:
```suggestion
    "appID": 680015438,
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+### Sequence Diagrams
+
+|||
+[UnsubscribeButton](./assets/UnsubscribeButton.png)
+|||
+
+### JSON Message Examples
+
+### Example Request
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:50]:
```suggestion
#### Example Request
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  "jsonrpc": "2.0",
+  "method": "Buttons.UnsubscribeButton",
+  "params": {
+    "appID": "680015438",
+    "buttonName": "VOLUME_UP"
+  }
+}
+```
+
+### Example Response
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:64]:
```suggestion
#### Example Response
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+  "id": 32,
+  "jsonrpc": "2.0",
+  "result": {
+    "code": 0,
+    "method": "Buttons.UnsubscribeButton"
+  }
+}
+```
+
+### Example Error
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:77]:
```suggestion
#### Example Error
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+### Example Request
+
+```json
+{
+  "id": 32,
+  "jsonrpc": "2.0",
+  "method": "Buttons.UnsubscribeButton",
+  "params": {
+    "appID": "680015438",
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:58]:
```suggestion
    "appID": 680015438,
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+|buttonName|[Common.ButtonName](../../common/enums/#buttonname)|true||
+
+### Response
+
+#### Parameters
+
+This RPC has no additional parameter requirements.
+
+
+### Sequence Diagrams
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:42]:
```suggestion
### Sequence Diagrams
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+|Name|Type|Mandatory|Additional|
+|:---|:---|:--------|:---------|
+|appID|Integer|true||
+|buttonName|[Common.ButtonName](../../common/enums/#buttonname)|true||
+
+### Response
+
+#### Parameters
+
+This RPC has no additional parameter requirements.
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:39]:
```suggestion
This RPC has no additional parameter requirements
```

File: docs/Buttons/SubscribeButton/index.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+2. Respond to SDL within DefaultTimeout period after successful button subscription.
+!!!
+
+#### Parameters
+This RPC has no additional parameter requirements.
+
+### Sequence Diagrams
+
+|||
+[SubscribeButton](./assets/SubscribeButton.png)
```
Comment (iCollin) [inline docs/Buttons/SubscribeButton/index.md:39]:
```suggestion
SubscribeButton
![SubscribeButton](./assets/SubscribeButton.png)
```

File: docs/Buttons/UnsubscribeButton/index.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+
+### Sequence Diagrams
+
+|||
+[UnsubscribeButton](./assets/UnsubscribeButton.png)
```
Comment (iCollin) [inline docs/Buttons/UnsubscribeButton/index.md:45]:
```suggestion
UnsubscribeButton
![UnsubscribeButton](./assets/UnsubscribeButton.png)
```

File: config.json [hunk 15]
```diff
@@ -1,4 +1,4 @@
...
```
Comment (iCollin) [inline config.json:1]: I am seeing `Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)` after this line was change. Can you please revert this change?

**Review by iCollin — APPROVED** [review]
